rosbag2_transport/src/rosbag2_transport/record_options_from_node_params.cpp

```diff
@@ -0,0 +1,127 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "rosbag2_transport/qos.hpp"
+#include "rosbag2_transport/record_options.hpp"
+#include "rosbag2_transport/utils/param_utils.hpp"
+#include "rosbag2_transport/record_options_from_node_params.hpp"
+
+namespace rosbag2_transport
+{
+
+RecordOptions get_record_options_from_node_params(std::shared_ptr<rclcpp::Node> node)
+{
+  RecordOptions record_options{};
+  record_options.all = node->declare_parameter<bool>(
+    "all",
+    false);
+
+  record_options.is_discovery_disabled = node->declare_parameter<bool>(
+    "is_discovery_disabled",
+    false);
+
+  record_options.topics = node->declare_parameter<std::vector<std::string>>(
+    "topics",
+    std::vector<std::string>());
+
+  record_options.rmw_serialization_format = node->declare_parameter<std::string>(
+    "rmw_serialization_format",
+    "");
+
+  auto desc_tpi = param_utils::int_param_description(
+    "Topic polling interval (ms)",
+    1,
+    std::numeric_limits<int64_t>::max());
+  auto topic_polling_interval_ = node->declare_parameter<int64_t>(
+    "topic_polling_interval",
+    100,
+    desc_tpi);
+  record_options.topic_polling_interval = std::chrono::milliseconds{topic_polling_interval_};
+
+  record_options.regex = node->declare_parameter<std::string>(
+    "regex",
+    "");
+
+  record_options.exclude = node->declare_parameter<std::string>(
+    "exclude",
+    "");
+
+  record_options.node_prefix = node->declare_parameter<std::string>(
+    "node_prefix",
+    "");
+
+  record_options.compression_mode = node->declare_parameter<std::string>(
+    "compression_mode",
+    "");
+
+  record_options.compression_format = node->declare_parameter<std::string>(
+    "compression_format",
+    "");
+
+  auto desc_cqs = param_utils::int_param_description(
+    "Compression queue size (messages)",
+    1,
+    std::numeric_limits<int64_t>::max());
+  auto compression_queue_size_ = node->declare_parameter<int64_t>(
+    "compression_queue_size",
+    1,
+    desc_cqs);
+  record_options.compression_queue_size = static_cast<uint64_t>(compression_queue_size_);
+
+  auto desc_cts = param_utils::int_param_description(
+    "Compression threads",
+    0,
+    std::numeric_limits<int64_t>::max());
+  auto compression_threads_ = node->declare_parameter<int64_t>(
+    "compression_threads",
+    0,
+    desc_cts);
+  record_options.compression_threads = static_cast<uint64_t>(compression_threads_);
+
+  // TODO(roncapat)
+  // std::unordered_map<std::string, rclcpp::QoS> topic_qos_profile_overrides{};
```

Comment:
Need to declare parameter for the qos_profile_overrides_path and parse it as yaml file.
Analog of the Pyton code

          qos_profile_overrides = {}  # Specify a valid default
          if args.qos_profile_overrides_path:
              qos_profile_dict = yaml.safe_load(args.qos_profile_overrides_path)
              try:
                  qos_profile_overrides = convert_yaml_to_qos_profile(
                      qos_profile_dict)
              except (InvalidQoSProfileException, ValueError) as e:
                  return print_error(str(e))
Similar to how we are doing it for rosbag2_storage_sqlite3

std::vector<std::string> pragma_entries;
  try {
    auto key =
      io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY ? "read" : "write";
    YAML::Node yaml_file = YAML::LoadFile(storage_config_uri);
    pragma_entries = yaml_file[key]["pragmas"].as<std::vector<std::string>>();
  } catch (const YAML::Exception & ex) {
    throw std::runtime_error(
            std::string("Exception on parsing sqlite3 config file: ") +
            ex.what());
  }